cmake/siconos_find_common_dep.cmake

```diff
@@ -267,12 +267,6 @@ if(UNIX)
 endif()
 
 
-# SiconosConfig.h generation and include
-if(EXISTS ${CMAKE_SOURCE_DIR}/config.h.cmake)
-  configure_file(${CMAKE_SOURCE_DIR}/config.h.cmake
-    ${CMAKE_BINARY_DIR}/SiconosConfig.h)
-endif()
-
 set(${PROJECT_NAME}_LOCAL_INCLUDE_DIRECTORIES
```

Comment:
I moved the config generation to the end of CMake instead of the beginning, otherwise I could not add `SICONOS_USE_CXSPARSE`, which I define during `externals/CMakeLists.txt`.  It is more usual to generate config as the last step in a build system, no?  In any case we can debate but it wasn't obvious to me that config generation happens during `siconos_find_common_dep`.  Otherwise, SuiteSparse/CXSparse detection could happen in that file instead, but it seemed to make sense to me to do it in externals.